src/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+interface SourceMapV3 {
+  file?: string;
+  names: string[];
+  sourceRoot?: string;
+  sources: string[];
+  sourcesContent?: (string | null)[];
```

Comment:
dumb question on my part, does this mean this allows for an array of strings **and** and an array of nulls or is it a union where it allows for the array to have both string and null?